nexus/src/external_api/http_entrypoints.rs

```diff
@@ -2006,11 +2011,34 @@ async fn instance_serial_console_stream(
         project: query.project.clone(),
         instance: path.instance,
     };
-    let instance_lookup = nexus.instance_lookup(&opctx, instance_selector)?;
-    nexus
-        .instance_serial_console_stream(conn, &instance_lookup, &query)
-        .await?;
-    Ok(())
+    let mut client_stream = WebSocketStream::from_raw_socket(
+        conn.into_inner(),
+        WebSocketRole::Server,
+        None,
+    )
+    .await;
+    match nexus.instance_lookup(&opctx, instance_selector) {
+        Ok(instance_lookup) => {
+            nexus
+                .instance_serial_console_stream(
+                    client_stream,
+                    &instance_lookup,
+                    &query,
+                )
+                .await?;
+            Ok(())
+        }
+        Err(e) => {
+            let _ = client_stream
+                .close(Some(CloseFrame {
+                    code: CloseCode::Error,
+                    reason: e.to_string().into(),
+                }))
+                .await
+                .is_ok();
```

Comment:
turbo nit/question for my understanding: since we're throwing away the result here anyway, is the `is_ok` needed?

Reply:
I think I wrote this as a way of being more explicit that the `must_use` thing we're ignoring is a Result rather than, say, a Future... after [cargo clippy caught a `let _ = ` that hid that from a refactor a couple weeks ago](https://github.com/oxidecomputer/propolis/pull/393#discussion_r1191912693).